src/websocket_tracker_connection.cpp

```diff
@@ -174,13 +174,13 @@ void websocket_tracker_connection::send_pending()
 	auto [msg, callback] = std::move(m_pending.front());
 	m_pending.pop();
 
-	std::visit([&](auto const& m)
+	std::visit([callback = std::move(callback)](auto const& m)
 		{
 			// Update requester and store callback
 			if (callback.lock())
 			{
 				m_requester = callback;
-				m_callbacks[m.info_hash] = callback;
+				m_callbacks[m.info_hash] = std::move(callback);
```

Comment:
you should still be able to move `callback` here. since it's captured by *non const* reference